pkg/controller/cainjector/controller.go

```diff
@@ -203,8 +204,15 @@ func (r *genericInjectReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 		// don't requeue if we're just not found, we'll get called when the secret gets created
 		return ctrl.Result{}, dropNotFound(err)
 	}
+
 	owner := OwningCertForSecret(&secret)
-	if owner == nil || *owner != certName {
+	// convert the annotation certname into the shortened label version
+	expectedCertNameFromLabel, err := util.NameToLabel(certName.Name)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	expectedOwner := types.NamespacedName{Namespace: certName.Namespace, Name: expectedCertNameFromLabel}
+	if owner == nil || *owner != expectedOwner {
```

Comment:
With this change, we'll leave around 'orphaned' Order resources that no longer match the new selector format.

In order to properly migrate here, I think we need to address this `// TODO: don't use a label selector at all here, instead we can index orders by their ownerRef and query based on owner reference alone` a few lines above. By doing so, we can automatically clean up orders that use the old format label value without having to support a migration period where we will have to query for Order's that have either the unhashed *OR* hashed value.